### PR TITLE
cli: cosmetic improvements to the output

### PR DESCRIPTION
**Pull request by pieleric**

### CODE REVIEW

File: src/odemis/cli/main.py [hunk 1]
```diff
@@ -148,35 +148,67 @@ def print_component(comp, pretty=True, level=0):
...
+    for comp, subg in graph.items():
         # first print the root component
-        print_component(root, pretty, level)
+        print_component(comp, pretty, level)
+        print_component_graph(subg, pretty, level + 1)
+
+
+def build_graph_children(comps):
+    """
+    comps (set)
```
Comment (Sabrina85) [inline src/odemis/cli/main.py]: To make it complete, please add a description.

File: src/odemis/cli/main.py [hunk 2]
```diff
@@ -148,35 +148,67 @@ def print_component(comp, pretty=True, level=0):
     # * if detector, display .shape
     # * if actuator, display .axes
 
-def print_component_tree(root, pretty=True, level=0):
+
+def print_component_graph(graph, pretty=True, level=0):
     """
     Print all the components starting from the root.
-    root (Component): the component at the root of the tree
+    graph (dict Component -> dict ...): parent -> children
```
Comment (Sabrina85) [inline src/odemis/cli/main.py]: what do the dots stand for?

File: src/odemis/cli/main.py [hunk 3]
```diff
@@ -148,35 +148,67 @@ def print_component(comp, pretty=True, level=0):
...
-    try:
-        microscope = model.getMicroscope()
-    except Exception:
-        raise IOError("Failed to contact the back-end")
+    # Show the root first, and don't use it for the graph, because its "children"
+    # are actually "dependencies", and it'd multiple parents in the graph.
+    microscope = model.getMicroscope()
+    print_component(microscope, pretty)
+
+    subcomps = model.getComponents() - {microscope}
```
Comment (Sabrina85) [inline src/odemis/cli/main.py]: What is this line doing?

File: src/odemis/cli/main.py [hunk 4]
```diff
@@ -148,35 +148,67 @@ def print_component(comp, pretty=True, level=0):
...
+                    del graph[child]
+                lefts.remove(comp)
+
+        if lefts == prev_lefts:
+            logging.warning("Some components have children not in the graph: %s",
+                            ", ".join(c.name for c in lefts))
+            for comp in lefts:
+                graph[comp] = {}
+            break
+    return graph
```
Comment (Sabrina85) [inline src/odemis/cli/main.py]: I think it would help the next reader to add some comments whats happening in the code :)

**Comment by pieleric** [comment]
Updated the comments and docstring. Hopefully that should make the code clear enough...

File: src/odemis/cli/main.py (hunk 3 above)
Comment (pieleric) [reply src/odemis/cli/main.py]: Reorganised the code. Hopefully that will make it more obvious.

**Pull request merged**